### PR TITLE
Updating the maintainers and CODEOWNERS file

### LINKED ISSUE #610: Baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions
Follow https://github.com/opensearch-project/.github/issues/125 to baseline MAINTAINERS, CODEOWNERS, and external collaborator permissions.

Close this issue when:

1. [MAINTAINERS.md](MAINTAINERS.md) has the correct list of project maintainers.
2. [CODEOWNERS](CODEOWNERS) exists and has the correct list of aliases.
3. Repo permissions only contain individual aliases as collaborators with maintain rights, admin, and triage teams. 
4. All other teams are removed from repo permissions.

If this repo's permissions was already baselined, please confirm the above when closing this issue.

### PR DESCRIPTION
**Pull request by bbarani**
### Description
Updating the MAINTAINERS.md and CODEOWNERS file

### Issues Resolved
Closes #610 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,3 @@
 @bowenlan-amzn
-@Angie-Zhang
 @getsaurabh02
 @praveensameneni
```
Comment (SuZhou-Joe) [inline .github/CODEOWNERS:3]:
```suggestion
@SuZhou-Joe
```

**Comment by SuZhou-Joe** [comment]
Please add AOS Shanghai team into maintainers & code owners as well.

File: .github/CODEOWNERS (hunk 1 above)
Comment (dblock) [reply .github/CODEOWNERS:3]: This is the wrong format, should be a `* ...list`.

**Comment by dblock** [comment]
@SuZhou-Joe We'll sync to what's currently in MAINTAINERS and you can fix the lists to be actual maintainers on your own time, please.

**Review by dblock — APPROVED** [review]

File: MAINTAINERS.md [hunk 2]
```diff
@@ -7,6 +7,5 @@ This document contains a list of maintainers in this repo. See [opensearch-proje
 | Maintainer | GitHub ID                                         | Affiliation |
 |----------|---------------------------------------------------| ----------- |
 | Bowen Lan | [bowenlan-amzn](https://github.com/bowenlan-amzn) | Amazon      |
-| Angie Zhang | [Angie-Zhang](https://github.com/Angie-Zhang)     | Amazon      |
 | Saurabh Singh | [getsaurabh02](https://github.com/getsaurabh02)           | Amazon      |
 | Praveen Sameneni | [praveensameneni](https://github.com/praveensameneni)           | Amazon      |
```
[the remark this answers is not in the document]
Comment (bbarani) [reply MAINTAINERS.md:11]: Let's add new maintainers after the baselining exercise.

**Review by dblock — APPROVED** [review]

**Review by prudhvigodithi — APPROVED** [review]

**Pull request merged**